uprotocol_vsomeip/vsomeip_utransport.py

```diff
@@ -195,80 +208,39 @@ def _get_instance(self, entity: UEntity, entity_type: VSOMEIPType) -> vsomeip.vS
         if entity_name in self._instances:
             return self._instances[entity_name]
 
-    def _invoke_handler(self, message_type: int, _: int, __: int, data: bytearray, request_id: int) -> bytearray:
-        """
-        callback for RPC method to set Future
-        """
-        if message_type == vsomeip.vSOMEIP.Message_Type.REQUEST.value:
-            return
-
-        req_id = LongUuidSerializer.instance().serialize(self._requests[request_id].attributes.id)
-        parsed_message = UMessage(
-            payload=UPayload(
-                value=bytes(data),
-                format=UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY,
-            ),
-            attributes=self._requests[request_id].attributes,
-        )
-
-        if not self._futures[req_id].done():
-            self._futures[req_id].set_result(parsed_message)
-        else:
-            _logger.info("Future result state is already finished or cancelled")
-
-    def _on_event_handler(self, message_type: int, service_id: int, event_id: int, data: bytearray, _: int) -> None:
-        """
-        handle responses from service with callback to listener registered
-        """
-        if message_type == vsomeip.vSOMEIP.Message_Type.REQUEST.value:
-            return None
-
-        payload_data = bytes(data)
-        message = self._published[service_id][event_id]  # Note: is mem shared copy
-
-        message_payload = message.payload
-        if message.payload.value != payload_data:
-            hint = UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY
-            message_payload = UPayload(value=payload_data, hint=hint)
-
-        parsed_message = UMessage(payload=message_payload, attributes=message.attributes)
-
-        for _, listener in self._registers[service_id][event_id]:
-            if listener:
-                listener.on_receive(parsed_message)  # call actual callback now...
-        return None
-
-    def _on_method_handler(
+    def _on_event_handler(
         self,
         message_type: int,
         service_id: int,
-        method_id: int,
+        instance: int,
+        event_id: int,
         data: bytearray,
-        request_id: int,
+        _: int,
```

Comment:
what is this and why isn't it being used?

Reply:
Underlying Vsomeip python library callbacks have a callback method definition, one of the attributes is request_id = client+session_id
In case of event handling we don't do anything with this attribute as all we need is service_id, instance_id, event_id data, message_type